WikiDBUpdaterHttp/WikiDBUpdaterHttp.cs

```diff
@@ -0,0 +1,144 @@
+        {
+            //url schema is /WikiDBUpdaterHttp?date=2021-12-31&days=3&discard&language=de,fr
+            //get and validate options
+            string YYYYMMDD = req.Query["date"];
+            DateTime startDate;
+            if (YYYYMMDD == null)
+                return new BadRequestObjectResult("Missing parameter: date");
+            if (!validateDate(YYYYMMDD, out startDate))
+                return new BadRequestObjectResult("Bad starting date: " + YYYYMMDD);
+
+            int daysToGo = 1;
+            string daysToGoString = req.Query["days"];
+            if (daysToGoString != null)
+                if (!int.TryParse(daysToGoString, out daysToGo))
+                    return new BadRequestObjectResult("Bad days parameter: " + daysToGoString);
+
+
+            string[] languages = allLanguageCodes;
+            string languageStrings = req.Query["languages"];
+            if (languageStrings != null)
+                if (!validateLanguages(languageStrings, out languages))
+                    return new BadRequestObjectResult("Bad languages parameter: " + languageStrings);
+
+
+            string discardString = req.Query["discard"];
+            bool discardOldData = discardString != null ? true : false;
+
+            IDBClient dbClient = new AzureStorageClient(tableClient);
+
+            await DBUpdates.updateDatabase(startDate, daysToGo, discardOldData, languages, 
+                articleExceptions, httpClient, dbClient, log);
+            
+            return new OkObjectResult("Successfull execution");
+        }
+
+        /// <summary>
+        /// Parses the YYYY-MM-DD string into DateTime
+        /// </summary>
+        /// <param name="_yyyymmdd">Date in a YYYY-MM-DD format</param>
+        /// <param name="_date">out for a date</param>
+        /// <returns>true if date was correctly parsed</returns>
+        bool validateDate(string _yyyymmdd, out DateTime _date)
+        {
+            _date = DateTime.Now;
+            int year, month, day;
+            string[] dateArray = _yyyymmdd.Split('-');
+
+            if (dateArray.Length != 3)
+                return false;
+
+            if (!int.TryParse(dateArray[0], out year))
+                return false;
+            if (!int.TryParse(dateArray[1], out month))
+                return false;
+            if (!int.TryParse(dateArray[2], out day))
+                return false;
+
+            _date = new DateTime(year, month, day);
+            return true;
+        }
+
+       
+        /// <summary>
+        /// Validates list of languages in "","","" format
+        /// </summary>
+        /// <param name="_language">string in "","","", format</param>
+        /// <param name="_languageArray">out for an output array</param>
+        /// <returns>returns true if string parsed into array</returns>
+        bool validateLanguages(string _language, out string[] _languageArray)
+        {
+            _languageArray = _language.ToLower().Split(',');
```

Comment:
```suggestion
            _languageArray = _language.ToUpperInvariant().Split(',');
```
`String.ToLower()` and `String.ToUpper()` formats numbers for people. People in different locations, have a bunch of weird characters, that they treat differently. Those methods are going to behave differently depending computer language (Really, Culture) settings. 
For computers, we have `String.ToLowerInvariant()` instead, which always behaves the same (but ignores various changes only some humans expect while lowercaseing)

Its .NET it is recommended to use uppercase strings, instead of lowercase strings, when needing a normalized case.